packages/playwright-test/src/reporters/junit.ts

```diff
@@ -133,6 +133,94 @@ class JUnitReporter implements Reporter {
     };
     entries.push(entry);
 
+    // Xray Test Management supports testcase level properties, where additional metadata may be provided
+    // some annotations are encoded as value attributes, other as cdata content
+    const properties: XMLEntry = {
+      name: 'properties',
+      children: [] as XMLEntry[]
+    };
+
+    const textValueBasedAnnotations = ['test_id', 'test_key', 'test_summary', 'requirements'];
+    for (const annotationType of textValueBasedAnnotations) {
+      const annotation = test.annotations.find(annotation => {
+        return annotation.type === annotationType;
+      });
+      if (annotation !== undefined) {
+        const property: XMLEntry = {
+          name: 'property',
+          attributes: {
+            name: annotationType,
+            value: (annotation?.description ? annotation.description : '')
+          }
+        };
+        properties.children?.push(property);
+      }
+    }
+
+    const textContentBasedAnnotations = ['test_description'];
+    for (const annotationType of textContentBasedAnnotations) {
+      const annotation = test.annotations.find(annotation => {
+        return annotation.type === annotationType;
+      });
+      if (annotation !== undefined) {
+        const property: XMLEntry = {
+          name: 'property',
+          attributes: {
+            name: annotationType
+          },
+          text: annotation.description
+        };
+        properties.children?.push(property);
+      }
+    }
+
+    // attachments are optionally embed as base64 encoded content on inner elements
+    const embedAttachments = test.annotations.find(annotation => {
+      return annotation.type === 'embed_attachments_in_report';
+    });
+    if (embedAttachments?.description === 'true') {
+      const evidence: XMLEntry = {
+        name: 'property',
+        attributes: {
+          name: 'testrun_evidence'
+        },
+        children: [] as XMLEntry[]
+      };
+      for (const result of test.results) {
+        for (const attachment of result.attachments) {
+          let contents;
+          if (attachment.body) {
+            contents = attachment.body.toString('base64');
+          } else {
+            if (!attachment.path)
+              continue;
+            try {
+              const attachmentPath = path.relative(this.config.rootDir, attachment.path);
+              if (fs.existsSync(attachmentPath))
+                contents = fs.readFileSync(attachmentPath, { encoding: 'base64' });
+            } catch (e) {
+            }
+          }
+
+          if (contents) {
+            const item: XMLEntry = {
+              name: 'item',
+              attributes: {
+                name: attachment.name
+              },
+              text: contents
+            };
+            evidence.children?.push(item);
```

Comment:
When embedding attachments, we should skip the "reference attachments" logic on the line 245.

Reply:
I agree